src/analysis/convex/simplicial_complex/basic.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import analysis.convex.topology
+
+/-!
+# Simplicial complexes
+
+In this file, we define simplicial complexes in `𝕜`-modules. A simplicial complex is a collection
+of simplices closed by inclusion (of vertices) and intersection (of underlying sets).
+
+We model them by a downward-closed set of affine independent finite sets whose convex hulls "glue
+nicely", each finite set and its convex hull corresponding respectively to the vertices and the
+underlying set of a simplex.
+
+## Main declarations
+
+* `simplicial_complex 𝕜 E`: A simplicial complex in the `𝕜`-module `E`.
+* `simplicial_complex.vertices`: The zero dimensional faces of a simplicial complex.
+* `simplicial_complex.facets`: The maximal faces of a simplicial complex.
+
+## TODO
+
+Simplicial complexes can be generalized to affine spaces once `convex_hull` has been ported.
+-/
+
+open finset set
+
+variables {𝕜 E ι : Type*} [ordered_ring 𝕜] [add_comm_group E] [module 𝕜 E]
+
+variables (𝕜 E)
+
+/-- A simplicial complex in a `𝕜`-module is a collection of simplices which glue nicely together. -/
+@[ext] structure simplicial_complex :=
+(faces : set (finset E))
+(not_empty_mem : ∅ ∉ faces)
+(indep : ∀ {s}, s ∈ faces → affine_independent 𝕜 (coe : (s : set E) → E))
+(down_closed : ∀ {s t}, s ∈ faces → t ⊆ s → t ≠ ∅ → t ∈ faces)
+(inter_subset_convex_hull : ∀ {s t}, s ∈ faces → t ∈ faces →
+  convex_hull 𝕜 ↑s ∩ convex_hull 𝕜 ↑t ⊆ convex_hull 𝕜 (s ∩ t : set E))
+
+namespace simplicial_complex
+variables {𝕜 E} {K : simplicial_complex 𝕜 E} {s t : finset E} {x : E}
+
+/-- The underlying space of a simplicial complex is the union of its faces. -/
+def space (K : simplicial_complex 𝕜 E) : set E := ⋃ s ∈ K.faces, convex_hull 𝕜 (s : set E)
+
+instance : has_coe (simplicial_complex 𝕜 E) (set E) := ⟨simplicial_complex.space⟩
```

Comment:
This seems dubious to me for the reasons discussed elsewhere (the "space" of K is always written |K|, and membership  x ∈ K never means x ∈ |K|, whatever else it might mean).